src/lib/theme/theme.service.ts

```diff
@@ -57,30 +71,39 @@ export class VantageThemeService {
     return this._activeTheme;
   }
 
-  public applyLightTheme(): void {
-    this.applyTheme(VantageTheme.LIGHT);
+  public applyLightTheme(): VantageTheme {
+    return this.applyTheme(VantageTheme.LIGHT);
   }
 
-  public applyDarkTheme(): void {
-    this.applyTheme(VantageTheme.DARK);
+  public applyDarkTheme(): VantageTheme {
+    return this.applyTheme(VantageTheme.DARK);
   }
 
-  public toggleTheme(): void {
-    this._activeTheme === VantageTheme.DARK ? this.applyLightTheme() : this.applyDarkTheme();
+  public toggleTheme(): VantageTheme {
+    return this._activeTheme === VantageTheme.DARK ? this.applyLightTheme() : this.applyDarkTheme();
   }
 
   public map(mapObject: IVantageThemeMap, fallback?: any): Observable<any> {
     return this.activeTheme$.pipe(map((value: VantageTheme) => (value in mapObject ? mapObject[value] : fallback)));
   }
 
-  private applyTheme(theme: VantageTheme): void {
+  private applyTheme(theme: VantageTheme, saveSetting: boolean = true): VantageTheme {
     this._renderer2.removeClass(
       this._document.querySelector('html'),
       theme === VantageTheme.DARK ? VantageTheme.LIGHT : VantageTheme.DARK,
     );
-    localStorage.setItem(THEME_LOCAL_STORAGE_KEY, theme);
     this._renderer2.addClass(this._document.querySelector('html'), theme);
-    this._activeTheme = <VantageTheme>localStorage.getItem(THEME_LOCAL_STORAGE_KEY);
+
+    if (saveSetting) {
+      localStorage.setItem(THEME_LOCAL_STORAGE_KEY, theme);
+    }
+
+    return (this._activeTheme = theme);
+  }
+
+  private checkOSPreference(): VantageTheme {
+    // it should now be light-by-default
+    return window.matchMedia('(prefers-color-scheme: dark)').matches ? VantageTheme.DARK : VantageTheme.LIGHT;
```

Comment:
Would be nice to also also listen to prefers-color-scheme events
https://stackoverflow.com/questions/59621784/how-to-detect-prefers-color-scheme-change-in-javascript

Reply:
oh good idea. let me see what i can do here! might be able to tie in with the same storage even subscription in the constructor

Reply:
That would be awesome

Reply:
Works great thanks for the suggestion

![vantage-theme](https://user-images.githubusercontent.com/3837706/77589267-ba34fe00-6ec1-11ea-99a5-6d4c985a1ff3.gif)